src/Initializer/InputParameters.cpp

```diff
@@ -185,6 +188,38 @@ static void readModel(ParameterReader& baseReader, SeisSolParameters& seissolPar
   reader.warnUnknown();
 }
 
+static void readCubeGenerator(ParameterReader& baseReader, SeisSolParameters& seissolParams) {
+  auto reader = baseReader.readSubNode("cubegenerator");
+
+  seissolParams.cubeGenerator.parameters.cubeMinX = reader.readWithDefault("cubeminx", 6);
+  seissolParams.cubeGenerator.parameters.cubeMaxX = reader.readWithDefault("cubemaxx", 6);
+  seissolParams.cubeGenerator.parameters.cubeMinY = reader.readWithDefault("cubeminy", 6);
+  seissolParams.cubeGenerator.parameters.cubeMaxY = reader.readWithDefault("cubemaxy", 6);
+  seissolParams.cubeGenerator.parameters.cubeMinZ = reader.readWithDefault("cubeminz", 6);
+  seissolParams.cubeGenerator.parameters.cubeMaxZ = reader.readWithDefault("cubemaxz", 6);
+
+  seissolParams.cubeGenerator.parameters.cubeX = reader.readWithDefault("cubex", 2);
+  seissolParams.cubeGenerator.parameters.cubeY = reader.readWithDefault("cubey", 2);
+  seissolParams.cubeGenerator.parameters.cubeZ = reader.readWithDefault("cubez", 2);
+
+  // only x dimension has its number of partitions set to number of MPI processes
+  seissolParams.cubeGenerator.parameters.cubePx = seissol::MPI::mpi.size();
```

Comment:
Just to be clear—we slice in the X direction, and keep the rest as it is? That may not really be the most surface-reducing method—but anything beyond is good in a another PR, if needed.

Reply:
That's correct. I'd delay finding a better solution until the next PR